common/address.go

```diff
@@ -51,6 +52,7 @@ func ConvertRecoverToError(r interface{}) error {
 	}
 }
 
+// DecodeBtcAddress decodes a BTC address from a given string and chainID
```

Comment:
I'm wondering, could it make sense to define an enum `BTCAddressType` Taproot|legacy|etc.. that would be return to the function?
We can at least in the test assert that it is the right format that is parsed

Reply:
It is possible to return a `BTCAddressType` from the API, but it's would be good  that we stick to the paradigm set by the `btcutil.DecodeAddress` API. The API return an interface of ``btcutil.Address` and the users' code doesn't have to care the type of the address they're manipulating. As long as the string can be parsed, it's then a valid address.